lib/rewards/server/models/reward.js

```diff
@@ -0,0 +1,161 @@
+const { ObjectsItemsRewardsAnimations } = require('./objects_items_rewards_animations');
+const { sc } = require('@reldens/utils');
+const { ObjectsConst } = require('../../../objects/constants');
+
+class Reward // TODO Extender de modelo base
+{
+    id = 0;
+    objectId = 0;
+    itemId = 0;
+    modifierId = 0;
+    experience = -1;
+    dropRate = -1;
+    dropQuantity = -1;
+    isUnique = false;
+    wasGiven = false;
+    hasDropBody = false;
+    animationData = null;
+
+
+    constructor(param)
+    {
+        this.id = param.id;
+        this.objectId = param.objectId;
+        this.itemId = param.itemId;
+        this.modifierId = param.modifierId;
+        this.experience = param.experience;
+        this.dropRate = param.dropRate;
+        this.dropQuantity = param.dropQuantity;
+        this.isUnique = param.isUnique === 1;
+        this.wasGiven = param.wasGiven === 1;
+        this.hasDropBody = param.hasDropBody === 1;
+        this.animationData = param.animationData;
+    }
+
+    validate()
+    {
+        //TODO: Validate this object.
+    }
+
+    isWinningReward()
+    {
+        if (0 === this.dropRate) {
+            return false;
+        }
+        return Math.floor(Math.random() * 100) <= this.dropRate;
+    }
+
+
+    isValidReward()
+    {
+        if (0 === this.dropQuantity) {
+            return false;
+        }
+
+        if (this.isUnique && this.wasGiven) {
+            return false;
+        }
+
+        if (!this.isItemType() && !this.isModifierType() && !this.hasExperienceSet()) {
+            return false;
+        }
+
+        return true;
+    }
+
+    isItemType()
```

Comment:
as general note for all these methods:
- don't use strict null, even mode considering you are creating the property with INT = 0 
```
objectId = 0;
itemId = 0;
modifierId = 0;
```
- replace them by just `!this.itemId` or `!this.modifierId`

Reply:
ok with removing the null or replacing it with false so in the evaluation the value inside itemId gets coarsed into a boolean, but I prefer to keep the methods as they give better readability to the code.